### ISSUE TITLE
Power window model results

### ISSUE THREAD

**Issue by juodaspaulius** [juodaspaulius/clafer#20]
Update on the power window model. 

I am using this power window model below and ran through the compiler. Unfortunately I am not getting any instances.  Unsat core tool is not very helpful.

To me, none of the Clafer constraint appear to over-constrain the model.

Clafer Model:
https://gist.github.com/juodaspaulius/a0f2df51e5b49dd24249#file-power_window_full-cfr

Generated Alloy model:
https://gist.github.com/juodaspaulius/542dc6cc2b62514768c7#file-power_window_full_15-als

Running with max scope of 15 takes 76 seconds on my machine.

```
Executing "Run show for 15"
   Solver=minisatprover(jni) Bitwidth=0 MaxSeq=0 SkolemDepth=1 Symmetry=20
   18410093 vars. 35805 primary vars. 34907246 clauses. 75917ms.
   No instance found. Predicate may be inconsistent. 40975ms.
   . contains 6 top-level formulas. 39ms.
```

**Comment by wasowski** [comment]
Have you tried to minimize the example? (by trying to remove parts, until it becomes satisfiable?)

It is quite possible that there is a bug in the model.

How do I get the tool installed? And how do I run it?

**Comment by wasowski** [comment]
I am trying to play with this instance to debug what is going on.  Will be taking notes here. The first problem I noticed is  that State.next is now clashing with some other name (alloy analyzer reports a clash with integer/next).  To get your example (the alloy version, above) to compile, I renamed next to next1 everywhere.  Is this because of the new alloy I use? (but I use alloy 4.2 from 2012 ... freshly downloaded).  

Then I tried to run the analyzer, but it spent an enormous time generating the CNF, and then it crashed with out of memory error (I am doing this on a rather strong machine). 

I used the same command as above (run show for 15), which seems to have only take 76s for you and timeouts for me? Hints?

**Comment by wasowski** [comment]
I worked a bit more on the debugging.  There seem to be 4 constraints that are causing inconsistencies ("seem" because usually a constraint is not really inconsistent but a conjunction with some other is).  Some testing seems to indicate that the first one, in line 186 is independent of the other problem (constraints in lines 237-242; all numbers in the Alloy file).

The line 186 seems to encode the initial constraint from line 32 (in .cfr file, stopped state).  We think that the parameters to localFirst are wrong. It appears that  `localFirst[r_c0_winStates,c0_WinController,this]` should be used instead of `localFirst[r_c0_winStates,c0_winStates, this]`.  It seems that `c0_winStates` is the same thing as `this` in the latter causing the problem (it seems that `localFirst` wants a parent and a child).  Having said that we did try our correction and it did not help.  So your opinion would be appreciated :)

Otherwise, when I get to this again, I shall try to test on a simpler model:

``` a
   xor b
       initial c ?
       d ?
```

**Comment by juodaspaulius** [comment]
Your're right! `inital` get's wrong encoding. In the your simple model [output](https://gist.github.com/juodaspaulius/a04636194ba514bdbd0f#file-simple-initial-cfr-L13) on line 13 we should have:
`((one t0 : localFirst[r_c0_b, root, this] | some this.@r_c0_c.t0)) }`

Working on a fix.

**Comment by wasowski** [comment]
Hi Paulius, have you managed to commit this?

**Comment by juodaspaulius** [comment]
No, I started but never completed a fix for this issue. I am sorry about it and hope its not too late. I will get this fixed tonight.

**Comment by juodaspaulius** [comment]
`Initial` encoding issue is now fixed.

**Comment by wasowski** [comment]
Thanks a lot!
